src/trust_anchor.rs

```diff
@@ -53,30 +53,40 @@ impl<'a> TrustAnchor<'a> {
     fn from_v1_der(cert_der: untrusted::Input<'a>) -> Result<Self, Error> {
         // X.509 Certificate: https://tools.ietf.org/html/rfc5280#section-4.1.
         cert_der.read_all(Error::BadDer, |cert_der| {
-            der::nested(cert_der, der::Tag::Sequence, Error::BadDer, |cert_der| {
-                let anchor = der::nested(cert_der, der::Tag::Sequence, Error::BadDer, |tbs| {
-                    // The version number field does not appear in v1 certificates.
-                    lenient_certificate_serial_number(tbs)?;
+            der::nested(
+                cert_der,
+                der::Tag::Sequence,
+                Error::TrailingData("TrustAnchorV1::outer"),
```

Comment:
same feedback to consider here: `outer` -> `Certificate`, `inner` -> `TBSCertificate`.

If you don't like this style, then I think it still makes sense to rename `TrustAnchorV1::inner` to `TrustAnchorV1::TBS` to match the certificate parsing error you have already.

Reply:
Used `TrustAnchorV1` and `TrustAnchorV1::tbs_certifcate` for these.